### PR TITLE
Added functionality to use a different clipboard tool than xclip

### PR DESCRIPTION
**Pull request by jfeick**
Added two new items to the configuration file:

-  clip_tool
- clip_tool_arguments

Defaults are:
clip_tool = 'xclip'
clip_tool_arguments = ''

### CODE REVIEW

File: autopass [hunk 1]
```diff
@@ -99,13 +103,20 @@ DEFAULT_CONFIG = expand_config(
     open_browser: 'Alt+o',
     copy_otp: 'Alt+c'
   },
-  alt_delay: 0.5
+  alt_delay: 0.5,
+  clip_tool: 'xclip',
+  clip_tool_arguments: ''
```
Comment (repomaa) [inline autopass]: this should probably be an empty array by default

File: autopass [hunk 2]
```diff
@@ -99,13 +103,20 @@ DEFAULT_CONFIG = expand_config(
...
 
 CONFIG = expand_config(YAML.load_file(config_file))
 CONFIG.reverse_merge!(DEFAULT_CONFIG)
 CONFIG[:autotype_1] ||= CONFIG[:password_key]
 CONFIG[:autotype_2] ||= CONFIG[:username_key]
+if !CONFIG.clip_tool_arguments.empty?
+  CONFIG[:clip_cmdline] = CONFIG.clip_tool + ' ' + CONFIG.clip_tool_arguments
+else
+  CONFIG[:clip_cmdline] = CONFIG.clip_tool
+end
```
Comment (repomaa) [inline autopass]: then you can omit this

File: autopass [hunk 3]
```diff
@@ -222,7 +233,7 @@ class Entry
       xwininfo = `xwininfo -int`
       title = xwininfo.match(/Window id: \d+ "(?<title>.*)"$/)[:title]
       print "selected window: `#{title}` "
-      IO.popen('xclip', 'w+') do |io|
+      IO.popen(CONFIG.clip_cmdline, 'w+') do |io|
```
Comment (repomaa) [inline autopass]: and use `IO.popen([CONFIG.clip_tool, *CONFIG.clip_tool_arguments])` here

File: autopass (hunk 2 above)
Comment (repomaa) [reply autopass]: or replace with `CONFIG[:clip_cmdline] = [CONFIG.clip_tool, *CONFIG.clip_tool_arguments]`

File: autopass (hunk 1 above)
Comment (repomaa) [reply autopass]: splatting would also allow nil as default.

**Comment by jfeick** [comment]
@jreinert I find the syntax of the configuration file super awkward if it's defining arguments as array:

```
# clipboard program to use:
#   clip_tool: 'xclip'
clip_tool: 'xclip'
clip_tool_arguments: 
    - '-selection'
    - 'clipboard'
```

Most users are not really used to yaml syntax.
What about omitting clip_tool_arguments all together? It would suffice to only use `clip_tool` to specify the program plus additional arguments:

```
clip_tool: 'xclip -selection clipboard'
```

**Comment by repomaa** [comment]
`clip_tool_arguments: [-selection, clipboard]` would also be valid. But i'd be okay with omitting the args as well. If you think omitting them is the way to go, I'd rename the config option to `clip_command`. Last but not least, please squash your commits. Then I'm happy to merge! :)

**Comment by repomaa** [comment]
Oh and protip: add `fixes #18` to your commit message for the issue to be resolved automatically when your pr is merged :)

**Comment by jfeick** [comment]
@jreinert done

**Comment by repomaa** [comment]
Very good! Thank you!

**Pull request merged**